src/braket/circuits/translations.py

```diff
@@ -0,0 +1,201 @@
+from functools import reduce, singledispatch
+from typing import Union
+
+import braket.circuits.gates as braket_gates
+import braket.circuits.noises as noises
+import braket.circuits.result_types as ResultTypes
+import braket.ir.jaqcd.shared_models as models
+from braket.circuits import Instruction, Observable, observables
+from braket.default_simulator import KrausOperation
+from braket.default_simulator.noise_operations import (
+    AmplitudeDamping,
+    BitFlip,
+    Depolarizing,
+    GeneralizedAmplitudeDamping,
+    PauliChannel,
+    PhaseDamping,
+    PhaseFlip,
+    TwoQubitDephasing,
+    TwoQubitDepolarizing,
+)
+from braket.ir.jaqcd import (
+    AdjointGradient,
+    Amplitude,
+    DensityMatrix,
+    Expectation,
+    Probability,
+    Sample,
+    StateVector,
+    Variance,
+)
+from braket.ir.jaqcd.program_v1 import Results
+
+BRAKET_GATES = {
+    "i": braket_gates.I,
+    "h": braket_gates.H,
+    "x": braket_gates.X,
+    "y": braket_gates.Y,
+    "z": braket_gates.Z,
+    "cv": braket_gates.CV,
+    "cnot": braket_gates.CNot,
+    "cy": braket_gates.CY,
+    "cz": braket_gates.CZ,
+    "ecr": braket_gates.ECR,
+    "s": braket_gates.S,
+    "si": braket_gates.Si,
+    "t": braket_gates.T,
+    "ti": braket_gates.Ti,
+    "v": braket_gates.V,
+    "vi": braket_gates.Vi,
+    "phaseshift": braket_gates.PhaseShift,
+    "cphaseshift": braket_gates.CPhaseShift,
+    "cphaseshift00": braket_gates.CPhaseShift00,
+    "cphaseshift01": braket_gates.CPhaseShift01,
+    "cphaseshift10": braket_gates.CPhaseShift10,
+    "rx": braket_gates.Rx,
+    "ry": braket_gates.Ry,
+    "rz": braket_gates.Rz,
+    "swap": braket_gates.Swap,
+    "iswap": braket_gates.ISwap,
+    "pswap": braket_gates.PSwap,
+    "xy": braket_gates.XY,
+    "xx": braket_gates.XX,
+    "yy": braket_gates.YY,
+    "zz": braket_gates.ZZ,
+    "ccnot": braket_gates.CCNot,
+    "cswap": braket_gates.CSwap,
+    "gpi": braket_gates.GPi,
+    "gpi2": braket_gates.GPi2,
+    "ms": braket_gates.MS,
+    "unitary": braket_gates.Unitary,
+}
+
+
+@singledispatch
+def braket_noise_gate_to_instruction(noise: KrausOperation) -> Union[Instruction]:
+    raise TypeError(f"Operation {type(noise).__name__} not supported")
```

Comment:
Split this into two functions:
```python
def braket_noise_gate_to_instruction(noise: KrausOperation) -> Union[Instruction]:
    return _braket_noise_gate_to_instruction(noise: KrausOperation)

@singledispatch
def _braket_noise_gate_to_instruction(noise: KrausOperation) -> Union[Instruction]:
    raise TypeError(f"Operation {type(noise).__name__} not supported")
```
In general, make `singledispatch` functions private because they don't play well with doc rendering.